pkg/operator/ceph/pool/controller.go

```diff
@@ -147,6 +148,14 @@ func (r *ReconcileCephBlockPool) reconcile(request reconcile.Request) (reconcile
 		// If not, we should wait for it to be ready
 		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
 		if !cephBlockPool.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// disable RBD stats collection if cephBlockPool is being deleted
+			cephBlockPool.Spec.EnableRBDStats = false
```

Comment:
Instead of setting this to false, `configureRBDStats()` can detect that the pool is marked for deletion and remove it from the list, right?

Reply:
`configureRBDStats()` retrieves a list of `cephBlockPool`s with `cephBlockPool.Spec.EnableRBDStats = true` and enable stat for those. We don't remove anything, instead we override the old list with a new one.

So, here when we set false and call `configureRBDStats()`, it excludes this particular object from the list and consequently gets removed. This reduces code as we're already detecting deletion.

Reply:
On line 318 I see a check for `cephBlockPool.GetDeletionTimestamp().IsZero()` and only then will the pool be added to the list. Doesn't that already cover this case? In any case, setting this to false will be ignored since it's not persisted back to the API and the `configureRBDStats()` refreshes the list of pools from the API.

Reply:
ahh right. I'll verify it again.